core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -375,6 +378,35 @@ private RetryPolicy resolveAwsRetryPolicy(SdkClientConfiguration config) {
                                        .defaultRetryMode(config.option(SdkClientOption.DEFAULT_RETRY_MODE))
                                        .resolve();
         return AwsRetryPolicy.forRetryMode(retryMode);
+        // fixme This will be changed like this to pick the configured retry strategy
+        // if no retry policy is configured.
+        /*
+        RetryPolicy policy = config.option(SdkClientOption.RETRY_POLICY);
+
+        if (policy != null) {
+            if (policy.additionalRetryConditionsAllowed()) {
+                return AwsRetryPolicy.addRetryConditions(policy);
+            } else {
+                return policy;
+            }
+        }
+
+        // If we don't have a configured retry policy we will use the configured retry strategy instead.
+        return null;
+         */
+    }
+
+    private RetryStrategy<?, ?> resolveAwsRetryStrategy(SdkClientConfiguration config) {
```

Comment:
Can we avoid the Generic wildcard types by referring to the parent classes  ?

Reply:
This type takes the builder and the built types with very specific constraints that allows it to implement the `CopyableBuilder` for the builder and the `ToCopyableBuilder` for the built type and *also* extends it to have separated implementations of this. There's no specific type I can use as we have three different and interchangeable implementations, Legacy, Standard and Adaptive, and the only super type for these is `RetryStrategy`